From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Juan Sanchez <juanesf91@gmail.com>
Date: Thu, 30 Jul 2026 10:00:00 -0400
Subject: [PATCH] clk: sunxi-ng: a523 mcu: add bus_dsp clock gate

Add the bus_dsp clock gate at MCU CCU register 0x0100 BIT(1). This clock
gate controls the DSP's AHB slave interface, which is required for the
DSP to access system peripherals (UART, etc.) over the system bus.

Signed-off-by: Juan Sanchez <juanesf91@gmail.com>
---
 drivers/clk/sunxi-ng/ccu-sun55i-a523-mcu.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/clk/sunxi-ng/ccu-sun55i-a523-mcu.c b/drivers/clk/sunxi-ng/ccu-sun55i-a523-mcu.c
index 197844f0f..f3b917efd 100644
--- a/drivers/clk/sunxi-ng/ccu-sun55i-a523-mcu.c
+++ b/drivers/clk/sunxi-ng/ccu-sun55i-a523-mcu.c
@@ -199,6 +199,8 @@ static SUNXI_CCU_GATE_DATA(bus_dsp_msgbox_clk, "bus-dsp-msgbox",
 static SUNXI_CCU_GATE_DATA(bus_dsp_cfg_clk, "bus-dsp-cfg",
 			   apb, 0x06c, BIT(0), 0);
 
+static SUNXI_CCU_GATE_DATA(bus_dsp_clk, "bus-dsp", ahb, 0x0100, BIT(1), 0);
+
 static SUNXI_CCU_GATE_DATA(bus_npu_hclk, "bus-npu-hclk", ahb, 0x070, BIT(1), 0);
 static SUNXI_CCU_GATE_DATA(bus_npu_aclk, "bus-npu-aclk", ahb, 0x070, BIT(2), 0);
 
@@ -310,6 +312,7 @@ static struct ccu_common *sun55i_a523_mcu_ccu_clks[] = {
 	&bus_audio_codec_clk.common,
 	&bus_dsp_msgbox_clk.common,
 	&bus_dsp_cfg_clk.common,
+	&bus_dsp_clk.common,
 	&bus_npu_aclk.common,
 	&bus_npu_hclk.common,
 	&mcu_timer0_clk.common,
@@ -358,6 +361,7 @@ static struct clk_hw_onecell_data sun55i_a523_mcu_hw_clks = {
 		[CLK_BUS_MCU_AUDIO_CODEC]	= &bus_audio_codec_clk.common.hw,
 		[CLK_BUS_MCU_DSP_MSGBOX]	= &bus_dsp_msgbox_clk.common.hw,
 		[CLK_BUS_MCU_DSP_CFG]		= &bus_dsp_cfg_clk.common.hw,
+		[CLK_BUS_MCU_DSP]		= &bus_dsp_clk.common.hw,
 		[CLK_BUS_MCU_NPU_HCLK]		= &bus_npu_hclk.common.hw,
 		[CLK_BUS_MCU_NPU_ACLK]		= &bus_npu_aclk.common.hw,
 		[CLK_MCU_TIMER0]		= &mcu_timer0_clk.common.hw,
@@ -379,7 +383,7 @@ static struct clk_hw_onecell_data sun55i_a523_mcu_hw_clks = {
 		[CLK_MCU_PWM0]			= &mcu_pwm0_clk.common.hw,
 		[CLK_BUS_MCU_PWM0]		= &bus_mcu_pwm0_clk.common.hw,
 	},
-	.num	= CLK_BUS_MCU_PWM0 + 1,
+	.num	= CLK_BUS_MCU_DSP + 1,
 };
 
 static struct ccu_reset_map sun55i_a523_mcu_ccu_resets[] = {
